src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-display-code.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, EventEmitter, Input, OnChanges, Output, SimpleChanges } from '@angular/core';
+import { of } from 'rxjs';
+import { catchError, tap } from 'rxjs/operators';
+import { CodeEditorRepositoryFileService } from 'app/exercises/programming/shared/code-editor/service/code-editor-repository.service';
+
+@Component({
+    selector: 'jhi-code-editor-tutor-assessment-display-code',
+    templateUrl: './code-editor-tutor-assessment-display-code.component.html',
+    styles: ['pre { display: inline; border: none; background-color: white; border-radius: unset ; padding: 1px; font-size: 11px} table {font-size: 11px}'],
+})
+export class CodeEditorTutorAssessmentDisplayCodeComponent implements OnChanges {
+    @Input()
+    selectedFile: string;
+    @Output()
+    onError = new EventEmitter<string>();
+
+    fileSession: { [fileName: string]: { code: string; cursor: { column: number; row: number } } } = {};
+    studentCode: string;
+    studentCodePerLine: string[];
+
+    isLoading = false;
+
+    constructor(private repositoryFileService: CodeEditorRepositoryFileService) {}
+
+    /**
+     * @function ngOnChanges
+     * @desc New selectedFile      => load the file from the repository and opens it
+     * @param {SimpleChanges} changes
+     */
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.selectedFile && this.selectedFile) {
+            // Current file has changed
+            // Only load the file from server if there is nothing stored in the file session
+            if (this.selectedFile && !this.fileSession[this.selectedFile]) {
+                this.loadFile(this.selectedFile);
+            } else {
+                this.initEditorAfterFileChange();
+            }
+        }
+    }
+
+    /**
+     * Setup the component after a file change occurred.
+     **/
+    initEditorAfterFileChange() {
+        if (this.selectedFile && this.fileSession[this.selectedFile]) {
+            this.studentCode = this.fileSession[this.selectedFile].code;
+            this.studentCodePerLine = this.studentCode.split('\n');
```

Comment:
does this also work, when students use Windows line endings?
Can you double check how other code editors split lines?
This approach looks to simple to me and might be error prone, maybe there are even helper method that can do this in a better way

Reply:
also see e.g. https://stackoverflow.com/questions/5034781/js-regex-to-split-by-line
(however keep in mind that we do not want to remove empty lines, nor do we want to remove whitespace)

Reply:
I am using now `this.studentCodePerLine = this.studentCode.split(/\r?\n/);` too catch the windows endings. Alternatively we could set the ace editor to only allow unix line breaks with `editor.session.setNewlineMode('unix')`